wagtail/admin/templates/wagtailadmin/shared/dialog/dialog.html

```diff
@@ -0,0 +1,45 @@
+{% load wagtailadmin_tags i18n %}
+{% with id=title|cautious_slugify %}
+    <div
+        id="{{ id }}"
+        aria-labelledby="title-{{ id }}"
+        aria-hidden="true"
+        data-dialog
+        class="w-fixed w-inset-0 w-z-dialog w-flex aria-hidden:w-hidden w-p-4"
+    >
+        {# Overlay #}
+        <div data-a11y-dialog-hide class="w-fixed w-inset-0 w-bg-black/40">
+        </div>
+
+        {# Dialog #}
+        <div class="w-w-full w-max-w-2xl sm:w-w-[600px] w-m-auto w-z-dialog w-relative w-bg-white w-shadow-dialog w-rounded-md w-overflow-hidden w-animate-fade-in">
+
+            {# Close button #}
+            <button type="button" data-a11y-dialog-hide aria-label="{% trans 'Close dialog' %}" class="w-absolute w-inline-flex w-justify-center w-items-center w-top-2 w-right-2 w-w-12 w-h-12 w-bg-transparent w-p-0">
+                {% icon name='cross' class_name="w-w-4 w-h-4 w-text-grey-600" %}
+            </button>
+
+            {# Message #}
+            {% if message_status and message_heading %}
+                {% dialog_message status=message_status heading=message_heading description=message_description %}
+            {% endif %}
+
+            <div class="w-p-8 sm:w-p-12 md:w-px-20 w-max-h-dialog w-overflow-auto w-scrollbar-thin">
+                {# Title #}
+                <h2 class="w-h1 w-relative sm:w-ml-12 md:w-ml-0 w-mt-0 w-mb-1" id="title-{{ id }}">
+                    {% if icon_name %}{% icon name=icon_name class_name="w-hidden sm:w-block w-w-5 w-h-5 w-text-primary w-absolute w-top-1/2 -w-translate-y-1/2 -w-left-10" %}{% endif %}
+                    {{ title }}
+                </h2>
+
+                {# Subtitle #}
```

Comment:
Sorry @stevedya I think that this should stay as description. 

However I think we could also add a subtitle that appears within the title .

Reply:
I'm not so sure on adding something like that to the h2 because it's intent is to be used as the title. I'm also not sure what value this would provide, because if someone would like to add extra text to the end of the title they could append it to the string when passing `title` or use the provided `subtitle` attr. (Also trying to keep the template tag as lean as possible.)

I do think renaming description to subtitle was a welcomed changed because it is more inline to how the designs appear:
<img width="610" alt="Screen Shot 2022-05-10 at 11 50 17 PM" src="https://user-images.githubusercontent.com/25041665/167782679-b60ed8d7-728e-4a96-ac90-99cf9022c632.png">

Reply:
good point  - maybe skip the subtitle idea. 

I think description (what you had before) is more suitable for a Dom element outside the header element. 

This is because we kind of have this convention already in the page headers subtitle appears inside the h1 and description (when used) is outside the h1 in its own element. 

It also gives us the ability to add subtitle in the future if the need arises.